sky/provision/do/utils.py

```diff
@@ -0,0 +1,209 @@
+"""DigitalOcean API client wrapper for SkyPilot.
+
+Example usage of `pydo` client library was mostly taken from here:
+https://github.com/digitalocean/pydo/blob/main/examples/poc_droplets_volumes_sshkeys.py
+"""
+
+import os
+from typing import Any, Dict, List, Optional
+import urllib
+import uuid
+
+from sky import sky_logging
+from sky.adaptors import do
+from sky.provision import common
+from sky.provision.do import constants
+from sky.utils import common_utils
+
+logger = sky_logging.init_logger(__name__)
+
+POSSIBLE_CREDENTIALS_PATHS = [
+    os.path.expanduser(
+        '~/Library/Application Support/doctl/config.yaml'),  # OS X
+    os.path.expanduser(
+        os.path.join(os.getenv('XDG_CONFIG_HOME', '~/.config/'),
+                     'doctl/config.yaml')),  # Linux
+]
+INITIAL_BACKOFF_SECONDS = 10
+MAX_BACKOFF_FACTOR = 10
+MAX_ATTEMPTS = 6
+SSH_KEY_NAME = f'sky-key-{common_utils.get_user_hash()}'
```

Comment:
`SSH_KEY_NAME` refers to the name of the key as stored by the digital ocean API, not the actual name of the key file stored on machine doing `sky launch`. The actual keys are passed in via
```
ssh_key_id(config.authentication_config['ssh_public_key'])['fingerprint']
```
during instance creation with `config.authentication_config['ssh_public_key']` coming from `get_or_generate_keys` already.

`SSH_KEY_NAME = f'sky-key-{common_utils.get_user_hash()}'` exists so that if multiple people share the same account but are launching across different machines, there isn't a collision in the API around the SSH key. lmk what you think

Reply:
Sounds good! If that is the case, we could consider rename it to sth like `SSH_KEY_NAME_ON_DO` to reduce confusion ; )